test/multiple-user-principal-types.test.js

```diff
@@ -0,0 +1,485 @@
+      Promise.all([
+        OneUser.login(commonCredentials),
+        AnotherUser.login(commonCredentials),
+      ])
+      .spread(function(at1, at2) {
+        userFromOneModelTokens = at1;
+        userFromAnotherModelTokens = at2;
+        return OneUser.logout(userFromOneModelTokens.id);
+      })
+      .then(function() {
+        return Promise.all([
+          AccessToken.findById(userFromOneModelTokens.id),
+          AccessToken.findById(userFromAnotherModelTokens.id),
+        ]);
+      })
+      .spread(function(userFromOneModelTokens, userFromAnotherModelTokens) {
+        assert(!userFromOneModelTokens, 'accessToken should not exist after logging out user1');
+        assert(userFromAnotherModelTokens, 'accessToken should exist after logging out user1');
+        done();
+      })
+      .catch(done);
+    });
+  });
+
+  describe('Password Reset: ', function() {
+    describe('User.resetPassword(options)', function() {
+      var options = {
+        email: 'foo@bar.com',
+        redirect: 'http://foobar.com/reset-password',
+      };
+
+      it('should create a temp accessToken to allow a user to change password', function(done) {
+        OneUser.resetPassword({email: options.email})
+        .catch(function(err) {
+          assert(!err);
+        });
+
+        OneUser.once('resetPasswordRequest', function(info) {
+          assertGoodToken(info.accessToken, userFromOneModel);
+          info.accessToken.user.getAsync()
+          .then(function(user) {
+            assert.equal(user.id, userFromOneModel.id);
+            assert.equal(user.email, userFromOneModel.email);
+            done();
+          })
+          .catch(done);
+        });
+      });
+    });
+  });
+
+  describe('AccessToken (session) invalidation when changing email', function() {
+    var anotherUserFromOneModel;
+
+    it('should only impact related user', function(done) {
+      Promise.all([
+        OneUser.create({email: 'original@example.com', password: 'bar'}),
+      ])
+      .spread(function(u1, u2, u3) {
+        anotherUserFromOneModel = u1;
+        return Promise.all([
+          OneUser.login({email: 'original@example.com', password: 'bar'}),
+          OneUser.login(commonCredentials),
+          AnotherUser.login(commonCredentials),
+        ]);
+      })
+      .then(function() {
+        return anotherUserFromOneModel.updateAttributes({email: 'updated@example.com'});
+      })
+      .then(function() {
+        return Promise.all([
+          AccessToken.find({where: {userId: anotherUserFromOneModel.id,
+            principalType: OneUser.modelName}}),
+          AccessToken.find({where: {userId: userFromOneModel.id,
+            principalType: OneUser.modelName}}),
+          AccessToken.find({where: {userId: userFromAnotherModel.id,
+            principalType: AnotherUser.modelName}}),
+        ]);
+      })
+      .spread(function(anotherUserFromOneModelTokens, userFromOneModelTokens,
+      userFromAnotherModelTokens) {
```

Comment:
while i generally prefer a bit more verbose code to make things easily readable, your proposal is still rather explicit. OK

**update** : the order of AccessToken.find() is not assured, i need to add an order filter on principalType name to make it assured, so the expect(data) can randomly fail if the user with `principalType: 'AnotherUser'` is returned first

`return AccessToken.find({'order': 'principalType ASC'});`

Now i'm not that confident the code is easily readable.
thoughts?